### DISCUSSION TITLE
Preview Deployment Comments Frustrations

### DISCUSSION THREAD

**Discussion by KHamilton-dev** [vercel/community#1248] [Show & Tell]
My team and I have run into some frustrating situations with this new feature that have made us turn off the comments. It's a neat concept, but it needs some wrinkles ironed out.

1) the ❌ symbol for there being a comment on your deployment is the same as a failed deployment. It's irksome thinking a deployment failed because someone thinks some CSS should change. Why not a 🟡 or something similar? 

2) the comments should be toggle-able by deployment, not project. Having 'clean' deployments would be nice

3) guests (non-vercel members) should be allowed to leave comments. Having a way for clients to voice concerns regarding specific UI elements without having to screenshot would be very valuable

There are other small ones, but these are the main complaints.

**Comment by amyegan** [comment]
Thanks for sharing this feedback @KHamilton-dev! I forwarded it to the dev team so they can see the improvements you suggested 🙂

**Comment by becca** [comment]
@KHamilton-dev Thank you for taking the time to give us this feedback and I'm sorry your experience hasn't been stellar.

Great feedback on `#1`, thank you. 

On `#2`, how would you envision this working? Would you be okay toggling this on/off per deployment and then also handling the sharing at this level too? We wanted to minimize the friction to having this feature on given its value, open to your ideas here. 

Regarding `#3`, this is possible. Once you enable Sharing on the Preview Deployment you can share it with external collaborators. Those external collaborators simply need a Vercel account and then can comment on your Preview Deployment. We have this documented [here](https://vercel.com/docs/concepts/deployments/comments/how-comments-work#enable-comments-from-people-outside-your-team) if you would like to read more. Does this solve your concern?

**Comment by KHamilton-dev** [comment]
Just noticed the yellow circle is similar to the 'building' symbol, so that may not have been my best example.

Our main desire is to be able to show a client a preview development without it being cluttered with developer comments about styling, etc. We can already do that in the PR anyways. Having the comments/toolbar defaulted to 'off' and then being able to toggle them 'on' for a specific deployment to send to clients would solve this problem, but that leads to my next thought:

We knew about the collaborator option, but enabling them to do so without having to create a Vercel account would be nice -- telling a client they need to create an account with a new 'cloud platform as a service' company in order to leave comment could dissuade a lot of people. Especially those accustomed to Google Docs' ability for anyone with a link to edit.

**Comment by gdborton** [comment]
@KHamilton-dev appreciate the feedback on the account. Right now we're requiring accounts (just email verification really) to help stay in front of potential abuse issues. This prevents both anonymous troll comments, but also using someone else's email and spamming their email via comments.

Also to add slightly to what Becca has already stated, you can disable comments for specific branches via environment variable configuration (+ a redeploy), see docs here:
https://vercel.com/docs/concepts/deployments/comments/specialized-usage#enable-or-disable-comments-for-branches

**Comment by KHamilton-dev** [comment]
Being able to turn them on/off with custom branch preview environment variables is actually pretty useful if we wanted to demo the preview without showing the Vercel toolbar. Good to know.

Thanks for the assistance everyone!
